Paylike/Transaction.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Paylike;
+/**
+ * Class Transaction
+ * @package Paylike
+ * Handles transaction operations.
+ *
+ * @version    1.0.0
+ */
+class Transaction {
+
+	/**
+	 * @param $transaction_id
+	 *
+	 * @return int|mixed
+	 * Return the transaction data
+	 *
+	 */
+	public static function fetch( $transaction_id ) {
+		$adapter = Client::getAdapter();
+		if ( ! $adapter ) {
+			return - 1;
+		}
+
+		return $adapter->request( 'transactions/' . $transaction_id );
+
+	}
+
+
+	/**
+	 * @param $transaction_id
+	 * The authorization takes place via the client script,
+	 * so this actually returns the transaction data
+	 *
+	 * @return int|mixed
+	 */
+	public static function authorize( $transaction_id ) {
```

Comment:
I don't think this method makes sense on this level, which is mostly a mapping from the HTTP API.
Also during a regular payment flow, this would not be enough to verify the payment anyway.